src/Test/L0/Listener/Configuration/ConfigurationManagerL0.cs

```diff
@@ -249,7 +249,7 @@ private TestHostContext CreateTestContext([CallerMemberName] String testName = "
                 Assert.True(s.AgentName.Equals(_expectedAgentName));
                 Assert.True(s.PoolId.Equals(3));
                 Assert.True(s.WorkFolder.Equals(_expectedWorkFolder));
-                Assert.True(s.MachineGroupName.Equals(_expectedMachineGroupName));
+                Assert.True(s.MachineGroupId.Equals(0));
```

Comment:
I understood that this test was written this way because MachineGroup.Id has no public setter.
See how the similar test was written for Queue? If Queue.Id was public -  Go ahead and modify the MachineGroup WebAPI